manager/orchestrator/constraintenforcer/constraint_enforcer.go

```diff
@@ -83,10 +84,12 @@ func (ce *ConstraintEnforcer) rejectNoncompliantTasks(node *api.Node) {
 		log.L.WithError(err).Errorf("failed to list tasks for node ID %s", node.ID)
 	}
 
-	var availableMemoryBytes, availableNanoCPUs int64
+	available := &api.Resources{}
+
 	if node.Description != nil && node.Description.Resources != nil {
-		availableMemoryBytes = node.Description.Resources.MemoryBytes
-		availableNanoCPUs = node.Description.Resources.NanoCPUs
+		available = node.Description.Resources.Copy()
+	} else {
+		available.ThirdParty = make(map[string]*api.ThirdPartyResource)
```

Comment:
Let's move this to an `if available.ThirdParty == nil` conditional, so it also covers the case where `available` was copied from `node.Description.Resources`.